### PR TITLE
Convert to Gradle and add download classes

### PR DESCRIPTION
**Pull request by psiinon**
Note that I've moved the webdrivers to a files directory to keep things cleaner - that ok?

### CODE REVIEW

File: .classpath [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (thc202) [inline .classpath]: The IDE specific files could be removed.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+/bin/
```
Comment (thc202) [inline .gitignore]: Could be something like [zap-api-java .gitignore](https://github.com/zaproxy/zap-api-java/blob/develop/.gitignore)? (except for subprojects entries which are not required in this case)

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,46 @@
+task wrapper(type: Wrapper) {
+    gradleVersion = '3.0'
```
Comment (thc202) [inline build.gradle]: The version could be set to 3.3.

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+apply plugin: 'java'
+
+buildDir = 'buildGradle'
```
Comment (thc202) [inline build.gradle]: It can use the default build directory in this case.

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+task downloadIeWebdrivers(type: JavaExec, dependsOn: 'classes') {
+    group 'ZAP'
+    description 'Downloads the IE Webdrivers.'
+    classpath configurations.runtime, sourceSets.main.output
+    main = 'org.zaproxy.libs.WebdriverDownloadIE'
+}
+
+task downloadAllWebdrivers(dependsOn: [downloadGeckoWebdrivers, downloadChromeWebdrivers, downloadIeWebdrivers]) {}
```
Comment (thc202) [inline build.gradle]: Could also have group/description.

File: settings.gradle [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+ * Detailed information about configuring a multi-project build in Gradle can be found
+ * in the user guide at https://docs.gradle.org/3.3/userguide/multi_project_builds.html
+ */
+
+/*
+// To declare projects as part of a multi-project build use the 'include' method
+include 'shared'
+include 'api'
+include 'services:webservice'
+*/
```
Comment (thc202) [inline settings.gradle]: The above comments could be removed.

**Review by thc202 — COMMENTED** [review]
The geckodriver binaries are not the same, different versions?

**Comment by psiinon** [comment]
Copied the original binaries and hopefully fixed those files.
Looks like there is a new version of the IEDriverServer but I'll update that in a new PR once this is merged.

File: src/main/java/org/zaproxy/libs/WebdriverDownloadIE.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+     * @param args
+     */
+    public static void main(String[] args) {
+
+        // IEDriverServer releases: http://www.seleniumhq.org/download/
+
+        final String VERSION = "3.0.0";
+
+        DownloadTools.downloadDriver(
+                "http://selenium-release.storage.googleapis.com/3.0/IEDriverServer_Win32_" + VERSION + ".zip",
```
Comment (thc202) [inline src/main/java/org/zaproxy/libs/WebdriverDownloadIE.java]: The `3.0` should also be extracted, that also changes between versions.

File: settings.gradle [hunk 8]
```diff
@@ -0,0 +1,11 @@
+/*
+ * This settings file was generated by the Gradle 'init' task.
+ *
+ * The settings file is used to specify which projects to include in your build.
+ * In a single project build this file can be empty or even removed.
+ *
+ * Detailed information about configuring a multi-project build in Gradle can be found
+ * in the user guide at https://docs.gradle.org/3.3/userguide/multi_project_builds.html
+ */
```
Comment (thc202) [inline settings.gradle]: This comment can also be deleted

**Comment by psiinon** [comment]
Changed those files.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thanks!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**